tests/suma/pxe_formula.pm

```diff
@@ -0,0 +1,68 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Installation and configuration of SUSE Manager pxe salt formula
+# Maintainer: Pavel Sladek <psladek@suse.cz>
+
+use base "sumatest";
+use 5.018;
+use testapi;
+use lockapi;
+use selenium;
+
+sub run {
+    my ($self) = @_;
+    $self->register_barriers('pxe_formula', 'pxe_formula_finish');
+
+    my $srvdir = get_var('SERVER_DIR');
+
+    if (check_var('SUMA_SALT_MINION', 'branch')) {
+        $self->registered_barrier_wait('pxe_formula');
+        $self->registered_barrier_wait('pxe_formula_finish');
+    }
+    elsif (check_var('SUMA_SALT_MINION', 'terminal')) {
```

Comment:
how about combining both if's? e.g. `if (get_var('SUMA_SALT_MINION') =~ /branch|terminal/)`

Reply:
I would rather keep this separated, mainly for convenience. Formula tests are scheduled on all SUTs and basically are following the same pattern - check for terminal, terminal2, branch or else it is master. 'minion' and 'branch' are different machines with different roles so if any test is added to either one of them, this tests will be definitely different for terminal and branch minions.